core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -602,7 +602,8 @@ class KafkaServer(val config: KafkaConfig, time: Time = SystemTime, threadNamePr
                                       maxMessageSize = config.messageMaxBytes,
                                       maxIoBytesPerSecond = config.logCleanerIoMaxBytesPerSecond,
                                       backOffMs = config.logCleanerBackoffMs,
-                                      enableCleaner = config.logCleanerEnable)
+                                      enableCleaner = config.logCleanerEnable,
+                                      brokerCompressionType = config.compressionType)
```

Comment:
@junrao said in the issue that we should use the topic config, but it seems like you are using the broker config?